src/common/misc.c

```diff
@@ -322,15 +322,17 @@ daos_sgl_get_bytes(d_sg_list_t *sgl, bool check_buf, struct daos_sgl_idx *idx,
 	daos_size_t len;
 
 	if (p_buf_len != NULL)
-	*p_buf_len = 0;
+		*p_buf_len = 0;
 
 	if (idx->iov_idx >= sgl->sg_nr)
 		return true; /** no data in sgl to get bytes from */
 
 	len = check_buf ? sgl->sg_iovs[idx->iov_idx].iov_buf_len :
 		sgl->sg_iovs[idx->iov_idx].iov_len;
 
-	D_ASSERT(idx->iov_offset < len);
+	D_ASSERT(idx->iov_offset <= len);
+	if (len == 0)
+		goto next_iov;
```

Comment:
so we choose to advance the iov_idx but don't set p_buf and return 0 p_buf_len to caller? and caller will have to call into this function again? I'm not sure if caller can handle this well or not, but it looks to me it's not necessary to implement such function in this way. (I think we'd always return a non-zero buf_len until reach the end)

Reply:
now the only user is daos_sgl_processor and do memcpy, for the case of 0 len it will set the buf_len = 0, so the memcpy will do nothing. that seems fine?

Reply:
yes, if caller can handle this, it won't cause error. my point is why we want to make this function to return a 0 buf_len?